### PR TITLE
Added search quadruplet algorithm

### PR DESCRIPTION
**Pull request by Caesarsage**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by cclauss — APPROVED** [review]

File: searches/search_quadruplet.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+def search_quadruplets(sequence: list[int], target: int) -> list[int]:
```
Comment (tianyizheng02) [inline searches/search_quadruplet.py:1]:
```suggestion
def search_quadruplets(sequence: list[int], target: int) -> list[list[int]]:
```
Return type is incorrect

File: searches/search_quadruplet.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    >>> search_quadruplets([2, 0, -1, 1, -2, 2], 2)
+    [-2, 0, 2, 2], [-1, 0, 1, 2]
+    Explanation: Both the quadruplets add up to the target.
+
+    Technique: Two Pointers
+    This problem follows the Two Pointers pattern and
+    shares similarities with Triplet Sum to Zero.
+
+    """
```
Comment (tianyizheng02) [inline searches/search_quadruplet.py:16]:
```suggestion
    """
    [Explain the problem here]
    Technique: Two Pointers
    This problem follows the Two Pointers pattern and
    shares similarities with Triplet Sum to Zero.

    >>> search_quadruplets([4, 1, 2, -1, 1, -3], 1)
    [-3, -1, 1, 4], [-3, 1, 1, 2]
    >>> search_quadruplets([2, 0, -1, 1, -2, 2], 2)
    [-2, 0, 2, 2], [-1, 0, 1, 2]
    """
```
Please reformat the docstring and explain the problem.

File: searches/search_quadruplet.py
Comment (tianyizheng02) [inline searches/search_quadruplet.py:1]: What if there are duplicate quadruplets? Ex: `[0, 0, 0, 0, 0]` and `0` as input

Comment (tianyizheng02) [reply searches/search_quadruplet.py:1]: Note that I ran this file locally and this file breaks on the above example: it returns the empty list, which is obviously  incorrect.

**Review by tianyizheng02 — COMMENTED** [review]
This solution is incorrect because it breaks when there are repeated numbers in the list. For example,
```py
>>> search_quadruplets([0, 0, 0, 0], 0)
>>> search_quadruplets([0, 0, 0, 0, 0], 0)
>>> search_quadruplets([0, 0, 0, 1], 1)
```
all return an empty list.

**Comment by tianyizheng02** [comment]
Thanks for your contribution, but I'll have to close this PR because your algorithm produces incorrect outputs. We're currently in the process of clearing out our backlog of open PRs, so we unfortunately don't have the time to significantly fix PRs that aren't already correct.

**Pull request closed without merging**